### PR TITLE
Sql joins

### PR DESCRIPTION
**Pull request by ahuang11**
image

### CODE REVIEW

File: lumen/ai/agents.py [hunk 1]
```diff
@@ -740,16 +765,7 @@ async def answer(self, messages: list | str):
...
-
-        # TODO: find a better way to replace for all tables
-        for table in memory.get("closest_tables", [table]):
-            sql_expr = source.get_sql_expr(table)
-            if f"FROM {table}" in sql_out:
-                sql_in = sql_expr.replace("SELECT * FROM ", "").replace(
-                    "SELECT * from ", ""
-                )
-                sql_out = sql_out.replace(f"FROM {table}", f"FROM {sql_in}")
-            memory["current_sql"] = sql_out
```
Comment (ahuang11) [inline lumen/ai/agents.py:752]: The template instructs it to use read_parquet if necessary.

File: lumen/ai/agents.py [hunk 2]
```diff
@@ -161,17 +161,25 @@ async def _system_prompt_with_context(
...
         else:
             schema = source.get_schema(table, limit=100)
         schema = dict(schema)
-        for field, spec in schema.items():
-            if "inclusiveMinimum" in spec:
-                spec["min"] = spec.pop("inclusiveMinimum")
-            if "inclusiveMaximum" in spec:
-                spec["max"] = spec.pop("inclusiveMaximum")
+
+        if include_min_max:
```
Comment (ahuang11) [inline lumen/ai/agents.py:171]: Just for saving tokens if min/max isn't needed.

File: lumen/ai/agents.py (hunk 1 above)
Comment (philippjfr) [reply lumen/ai/agents.py:752]: This is great!

**Review by philippjfr — APPROVED** [review]
Amazing, great work! Exactly what I thought we would need.

**Pull request merged**